internal/cluster/distribution/eks/eksprovider/workflow/amazon.go

```diff
@@ -483,15 +483,28 @@ func WaitForASGToBeFulfilled(
 					if autoscaling.IsErrorFinal(err) {
 						return errors.WithDetails(err, "nodePoolName", nodePoolName, "stackName", aws.StringValue(asGroup.AutoScalingGroupName))
 					}
-					// log.Debug(err)
 					continue
 				}
 				if ok {
-					// log.Debug("ASG is healthy")
+					logger.Debug("ASG is healthy")
 					return nil
 				}
 			} else {
-				return errors.Errorf("waiting for ASG to be fulfilled timed out after %d x %s", asgFulfillmentWaitAttempts, asgFulfillmentWaitInterval)
+				asGroup, err := m.GetAutoscalingGroupByStackName(stackName)
+				if err != nil {
+					return errors.Errorf("waiting for ASG to be fulfilled timed out after %d x %s, could not get ASG in stack: %s", asgFulfillmentWaitAttempts, asgFulfillmentWaitInterval, stackName)
+				}
+				lastActivity, err := m.LastAutoscalingActivity(ctx, asGroup.AutoScalingGroupName)
+				if lastActivity != nil {
+					return errors.Errorf(
+						"waiting for ASG to be fulfilled timed out after %d x %s, status: %s, cause: %s, description: %s",
+						asgFulfillmentWaitAttempts,
+						asgFulfillmentWaitInterval,
+						aws.StringValue(lastActivity.StatusCode),
+						aws.StringValue(lastActivity.Cause),
+						aws.StringValue(lastActivity.Description),
+					)
+				}
```

Comment:
**Recommended**: If there is no reason to make the `internal/cluster/distribution/eks/ekscluster/nodepools/wait_for_asg.go` and `internal/cluster/distribution/eks/eksprovider/workflow/amazon.go` versions different, I would propose to use the exact same code for the 2.

(I'm not sure why there are 2 different implementations, I want to deduplicate those in the future if possible in a different PR, but until it's done I would prefer keeping them in sync unless required not to.)

```suggestion
				asGroup, err := m.GetAutoscalingGroupByStackName(asgName)
				if err != nil {
					return errors.Errorf("waiting for ASG to be fulfilled timed out after %d x %s, could not get ASG in stack: %s", waitAttempts, waitInterval, asgName)
				}

				lastActivity, err := m.LastAutoscalingActivity(ctx, asGroup.AutoScalingGroupName)
				if err != nil {
					return errors.Errorf("waiting for ASG to be fulfilled timed out after %d x %s, could not get last ASG activity in stack: %s", waitAttempts, waitInterval, asgName)
				}

				return errors.Errorf(
					"waiting for ASG to be fulfilled timed out after %d x %s, status: %s, cause: %s, description: %s",
					waitAttempts,
					waitInterval,
					aws.StringValue(lastActivity.StatusCode),
					aws.StringValue(lastActivity.Cause),
					aws.StringValue(lastActivity.Description),
				)
```

Reply:
Apologies for not being clear enough in my previous comment, I thought it is obvious I meant to make all the changes here as well as done there, not just the `err` assignment.

Reply:
Sorry, I just keep forgetting that this code is present in two places...

Reply:
Yeah, I advise you to not rush the different tasks, because context switching is difficult to be done right.

Take your time, this is the chance to learn how things work, don't rush it, there is no need.